### PR TITLE
Test login

### PR DESCRIPTION
**Pull request by rap0so**

### CODE REVIEW

File: src/__tests__/Login.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event'
+
+import { Provider } from 'react-redux';
+import store from '../app/store';
+import App from '../App';
+import GlobalContextProvider from '../app/contexts';
+import * as firebaseMock from '../app/firebase-configs';
```
Comment (Mazuh) [inline src/__tests__/Login.test.js:9]: this is not a mock, it's the real module, even if you're planning to intercept it with `spyOn` later.

**Review by Mazuh — CHANGES_REQUESTED** [review]
It's just 1 important review and another cosmetic one.

File: src/__tests__/Login.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <Provider store={store}>
+          <GlobalContextProvider>
+            <App />
+          </GlobalContextProvider>
+        </Provider>
+      )
+  )
+  
+  it('should login', async () => {
+    const fakeSignIn = jest.spyOn(firebaseMock, 'signInWithEmailAndPassword')
```
Comment (Mazuh) [inline src/__tests__/Login.test.js:29]: (PS: you're spying internal code, this does not test the actual firebase integration, this test can still pass even if I uninstall firebase completely from the application. But it's fine by now, it's a good start.)

File: src/__tests__/Login.test.js (hunk 1 above)
Comment (rap0so) [reply src/__tests__/Login.test.js:9]: I'm solving this conversation with the caveat that it should be improved in the future.

File: src/__tests__/Login.test.js (hunk 2 above)
Comment (rap0so) [reply src/__tests__/Login.test.js:27]: I'm solving this conversation with the caveat that it should be improved in the future.

**Review by Mazuh — APPROVED** [review]

**Pull request merged**